src/replication/DealReplicationWorker.ts

```diff
@@ -55,10 +57,34 @@ export default class DealReplicationWorker extends BaseService {
 
   private async pollWork (): Promise<boolean> {
     this.logger.debug(`${this.workerId} - Polling for work`);
+    await this.checkCronChange();
     const hasDoneWork = await this.pollReplicationWork();
     return hasDoneWork;
   }
 
+  private async checkCronChange (): Promise<void> {
+    const activeCronWorks = await Datastore.ReplicationRequestModel.find({
+      cronSchedule: { $ne: null },
+      workerId: { $ne: null },
```

Comment:
this is checking cron works for all workers, but does it need to care about jobs for other worker? I think you can filter it to this worker Id.

Reply:
There is only one worker at any time... so it doesn't matter I guess?

Reply:
It needs to be scaled out in the future when possible.